### DISCUSSION TITLE
Demystifying Architectural Decision Records: Why Every Project Needs Them | Dan Does Code

### DISCUSSION THREAD

**Discussion by giscus** [danielmackay/nextjs-personal-blog#100] [Blog Comments]
# Demystifying Architectural Decision Records: Why Every Project Needs Them

In the world of software development, Architectural Decision Records (ADRs) are indispensable for capturing and communicating crucial architectural choices throughout a project's lifecycle. This blog post explores the significance of ADRs, emphasizing their role in providing historical context, promoting transparency and accountability, facilitating collaboration, and enabling informed decision-making. It also offers practical insights on creating and maintaining ADRs effectively. By embracing ADRs, software development teams can enhance project clarity, foster collaboration, and make more informed decisions, leading to successful and resilient projects.

https://www.dandoescode.com/blog/demystifying-architectural-decision-records-why-every-project-needs-them

**Comment by SamBWagner** [comment]
I really like that you showed your actual site implementing log4brains and how Decision Records can be integrated into a project. 

I do think that keeping decision records lean is important. So I would say that there are things in all the provided templates which make them feel verbose.

I think something as simple as:

---

## Decision - Using JavaScript

We've chosen to use JavaScript for our backend language as our team is very new and need something lightweight as the project is primarily frontend with some light data removal

---

Would be sufficient. Though of course is the decision is more complex. Just add more :P

**Reply by danielmackay** [reply]
@SamBWagner - I agree.  If the decision is straightforward (but still important), I would aim to keep the ADR lightweight, and only add more information where appropriate (i.e. complex decision, or differing opinions within the team).

At a minimum, I would like to always see the following in an ADR:

- Context and Problem Statement
- Decision Outcome

You can find an example of an ADR likes this I've created here: [20230604-use-domain-services-to-encapsulate-external-behavior](https://danielmackay.github.io/dotnet-domain-driven-design/log4brains/adr/20230604-use-domain-services-to-encapsulate-external-behavior/)

**Reply by danielmackay** [reply]
Also, if you look at the template I'm using for my [DDD](https://github.com/danielmackay/dotnet-domain-driven-design) repo, you'll see that most of the sections are indeed optional: [template](https://github.com/danielmackay/dotnet-domain-driven-design/blob/main/docs/decisions/template.md?plain=1)
